src/base/ApplicationECSContainerDefinition.ts

```diff
@@ -92,81 +35,66 @@ export interface ApplicationECSContainerDefinitionProps {
+      : null,
+    dnsServers: null,
+    mountPoints: [],
+    workingDirectory: null,
+    secrets: config.secretEnvVars ?? null, // env vars default is [], whereas secrets default is null. makes sense.
+    dockerSecurityOptions: null,
+    memory: null,
+    memoryReservation: config.memoryReservation ?? null,
+    volumesFrom: [],
+    stopTimeout: null,
+    image: config.containerImage,
+    startTimeout: null,
+    firelensConfiguration: null,
+    dependsOn: null,
+    disableNetworking: null,
+    interactive: null,
+    healthCheck: config.healthCheck ?? null,
+    essential: true,
+    links: null,
+    hostname: null,
+    extraHosts: null,
+    pseudoTerminal: null,
+    user: null,
+    readonlyRootFilesystem: false,
+    dockerLabels: null,
+    systemControls: null,
+    privileged: null,
+    name: config.name,
+  };
 
   if (config.command) {
-    templateInstance = templateInstance.replace(
-      '[[COMMAND]]',
-      '[' + config.command.map((c) => `"${c}"`).join(',') + ']'
-    );
-  } else {
-    templateInstance = templateInstance.replace('"command": [[COMMAND]],', '');
-  }
-
-  templateInstance = templateInstance.replace('[[LOG_GROUP]]', config.logGroup);
-  templateInstance = templateInstance.replace(
-    '[[HOST_PORT]]',
-    config.hostPort.toString()
-  );
-  templateInstance = templateInstance.replace(
-    '[[CONTAINER_PORT]]',
-    config.containerPort.toString()
-  );
-  templateInstance = templateInstance.replace(
-    '[[CONTAINER_IMAGE]]',
-    config.containerImage
-  );
-  templateInstance = templateInstance.replace('[[NAME]]', config.name);
-  templateInstance = templateInstance.replace(
-    '[[REPOSITORY_CREDENTIALS_PARAMETER]]',
-    config.repositoryCredentialsParam
-      ? `{"credentialsParameter": "${config.repositoryCredentialsParam}"}`
-      : 'null'
-  );
-
-  templateInstance = templateInstance.replace('[[ENV_VARS]]', envVarsValue);
-  templateInstance = templateInstance.replace(
-    '[[SECRET_ENV_VARS]]',
-    secretEnvVarsValue
-  );
-
-  templateInstance = templateInstance.replace(
-    '[[CPU]]',
-    config.cpu ? config.cpu.toString() : '0'
-  );
-
-  templateInstance = templateInstance.replace(
-    '[[MEMORY_RESERVATION]]',
-    config.memoryReservation ? config.memoryReservation.toString() : 'null'
-  );
-
-  templateInstance = templateInstance.replace(
-    '[[PROTOCOL]]',
-    config.protocol ?? 'tcp'
-  );
-
-  if (config.healthCheck) {
-    templateInstance = templateInstance.replace(
-      '[[HEALTH_CHECK]]',
-      JSON.stringify({
-        command: config.healthCheck.command,
-        interval: config.healthCheck.interval,
-        retries: config.healthCheck.retries,
-        startPeriod: config.healthCheck.startPeriod,
-        timeout: config.healthCheck.timeout,
-      })
-    );
-  } else {
-    templateInstance = templateInstance.replace('[[HEALTH_CHECK]]', 'null');
+    containerDefinition['command'] = config.command;
```

Comment:
Do we leave this out if not defined? Interesting if so.. Can we comment on this?

Reply:
Good catch, I should've called this out for review. I did this to keep the behavior consistent, but I don't know why it was that way originally. I looked at the history and AWS docs:
- This comment from yourself is relevant, but doesn't reveal why the decision was made: https://github.com/Pocket/terraform-modules/pull/36#discussion_r515252198
- [The AWS docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-containerdefinitions.html#cfn-ecs-taskdefinition-containerdefinition-command) do not specify whether an empty list has different behavior than omitting this parameter.

Reply:
So I found it odd, but I don't what comment to but there.

@jpetto, do you remember why `command` is removed from the json by default?